File: lexicaforge/php/LexicaForgeTest.php
<?php

namespace LexicaForge;

use PHPUnit\Framework\TestCase;

class LexicaForgeTest extends TestCase {
    private $nlp;

    protected function setUp(): void {
        $this->nlp = new LexicaForge();
    }

    public function testTokenize() {
        $text = "Hello World! This is a test.";
        $tokens = $this->nlp->tokenize($text);
        
        $this->assertCount(6, $tokens);
        $this->assertEquals("hello", $tokens[0]);
        $this->assertEquals("world!", $tokens[1]);
        $this->assertEquals("this", $tokens[2]);
        $this->assertEquals("is", $tokens[3]);
        $this->assertEquals("a", $tokens[4]);
        $this->assertEquals("test.", $tokens[5]);
    }

    public function testTokenizeEmptyString() {
        $text = "";
        $tokens = $this->nlp->tokenize($text);
        $this->assertEmpty($tokens);
    }

    public function testExtractEntities() {
        $text = "John works at Google and Mary is at Microsoft";
        $entities = $this->nlp->extractEntities($text);
        
        $this->assertCount(4, $entities);
        $this->assertContains("John", $entities);
        $this->assertContains("Google", $entities);
        $this->assertContains("Mary", $entities);
        $this->assertContains("Microsoft", $entities);
    }

    public function testExtractEntitiesNoMatches() {
        $text = "this is a test without proper nouns";
        $entities = $this->nlp->extractEntities($text);
        $this->assertEmpty($entities);
    }

    public function testCalculateSimilarity() {
        $text1 = "Hello World";
        $text2 = "Hello Universe";
        $similarity = $this->nlp->calculateSimilarity($text1, $text2);
        
        $this->assertEqualsWithDelta(0.5, $similarity, 0.001);
    }

    public function testCalculateSimilarityIdentical() {
        $text = "Hello World";
        $similarity = $this->nlp->calculateSimilarity($text, $text);
        $this->assertEqualsWithDelta(1.0, $similarity, 0.001);
    }

    public function testCalculateSimilarityNoCommon() {
        $text1 = "Hello World";
        $text2 = "Goodbye Universe";
        $similarity = $this->nlp->calculateSimilarity($text1, $text2);
        $this->assertEqualsWithDelta(0.0, $similarity, 0.001);
    }

    public function testCacheFunctionality() {
        $text = "Test caching";
        
        // First call should compute
        $tokens1 = $this->nlp->tokenize($text);
        
        // Second call should use cache
        $tokens2 = $this->nlp->tokenize($text);
        
        $this->assertSame($tokens1, $tokens2);
    }
} 

File: lexicaforge/java/LexicaForgeTest.java
package com.lexicaforge;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.util.List;

public class LexicaForgeTest {
    private final LexicaForge nlp = new LexicaForge();

    @Test
    public void testTokenize() {
        String text = "Hello World! This is a test.";
        List<String> tokens = nlp.tokenize(text);
        
        assertEquals(6, tokens.size());
        assertEquals("hello", tokens.get(0));
        assertEquals("world!", tokens.get(1));
        assertEquals("this", tokens.get(2));
        assertEquals("is", tokens.get(3));
        assertEquals("a", tokens.get(4));
        assertEquals("test.", tokens.get(5));
    }

    @Test
    public void testTokenizeEmptyString() {
        String text = "";
        List<String> tokens = nlp.tokenize(text);
        assertTrue(tokens.isEmpty());
    }

    @Test
    public void testExtractEntities() {
        String text = "John works at Google and Mary is at Microsoft";
        List<String> entities = nlp.extractEntities(text);
        
        assertEquals(4, entities.size());
        assertTrue(entities.contains("John"));
        assertTrue(entities.contains("Google"));
        assertTrue(entities.contains("Mary"));
        assertTrue(entities.contains("Microsoft"));
    }

    @Test
    public void testExtractEntitiesNoMatches() {
        String text = "this is a test without proper nouns";
        List<String> entities = nlp.extractEntities(text);
        assertTrue(entities.isEmpty());
    }

    @Test
    public void testCalculateSimilarity() {
        String text1 = "Hello World";
        String text2 = "Hello Universe";
        double similarity = nlp.calculateSimilarity(text1, text2);
        
        assertEquals(0.5, similarity, 0.001);
    }

    @Test
    public void testCalculateSimilarityIdentical() {
        String text = "Hello World";
        double similarity = nlp.calculateSimilarity(text, text);
        assertEquals(1.0, similarity, 0.001);
    }

    @Test
    public void testCalculateSimilarityNoCommon() {
        String text1 = "Hello World";
        String text2 = "Goodbye Universe";
        double similarity = nlp.calculateSimilarity(text1, text2);
        assertEquals(0.0, similarity, 0.001);
    }

    @Test
    public void testCacheFunctionality() {
        String text = "Test caching";
        
        // First call should compute
        List<String> tokens1 = nlp.tokenize(text);
        
        // Second call should use cache
        List<String> tokens2 = nlp.tokenize(text);
        
        assertSame(tokens1, tokens2);
    }
} 